include/aikido/constraint/outcome/CollisionFreeOutcome.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef AIKIDO_CONSTRAINT_COLLISIONFREEOUTCOME_HPP_
+#define AIKIDO_CONSTRAINT_COLLISIONFREEOUTCOME_HPP_
+
+#include <vector>
+#include "TestableOutcome.hpp"
+#include "dart/collision/CollisionObject.hpp"
+#include "dart/collision/Contact.hpp"
+#include "dart/dynamics/BodyNode.hpp"
+#include "dart/dynamics/ShapeFrame.hpp"
+#include "dart/dynamics/ShapeNode.hpp"
+
+namespace aikido {
+namespace constraint {
+
+class CollisionFreeOutcome : public TestableOutcome
+{
+public:
+  /// Documentation inherited.
+  bool isSatisfied() const override;
+
+  /// Returns a string with each pair of CollisionObject names on a separate
+  /// line. Each pair is also marked as being a normal collision or self
+  /// collision.
+  std::string toString() const override;
+
+  /// Store a Contact object from a pairwise collision. Used by CollisionFree
+  // to modify this object.
+  /// \param pairwiseContact Contact object to store.
```

Comment:
Nit: We prefer mark whether the parameter is input param or output param using `[in]` and `[out]` affixes to `\param`. Please change this to `/// \param[in] pairwiseContact` (also other parameters in this header).